skypy/power_spectrum/_camb.py

```diff
@@ -98,4 +110,9 @@ def camb(wavenumber, redshift, cosmology, A_s, n_s):
                                                    maxkh=np.max(k_h.value),
                                                    npoints=len(k_h.value))
 
-    return pzk[redshift_order[::-1]].T
+    if len(redshift_order) > 1:
+        power_output = pzk[redshift_order[::-1]]
+    else:
+        power_output = pzk[redshift_order[::-1]].reshape(return_shape)
+
+    return power_output
```

Comment:
Why is this not just `return pzk[redshift_order[::-1]]` i.e. as before without the transposition?

Reply:
let me check

Reply:
If I just do 
`return pzk[redshift_order[::-1]]`
the shape of the power spectrum is `(1, 100)` for 100 values of `k` and NOT `(100,)`.
I am sure there is a more clever way to do it right. Do you have any suggestion?

Reply:
You can `np.squeeze` the result, although then you get shape `(100,)` even if redshift is an array with one entry. If you want to get it exactly right, you can do `.reshape(*np.shape(redshifts), *np.shape(wavenumber))` which is ugly but should correctly differentiate scalars and one-entry array input.

Reply:
Hmm that doesn't seem to work either.